src/transaction.ts

```diff
@@ -434,16 +430,32 @@ class Transaction extends DatastoreRequest {
     const callback =
       typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : cb!;
 
-    this.request_(
-      {
-        client: 'DatastoreClient',
-        method: 'rollback',
-        gaxOpts: gaxOptions || {},
+    if (this.state === TransactionState.EXPIRED) {
+      callback(new Error(transactionExpiredError));
+      return;
+    }
+    if (this.state === TransactionState.NOT_STARTED) {
```

Comment:
Just wondering what are the allowed states here?  NOT_TRANSACTION and IN_PROGRESS?

Reply:
NOT_TRANSACTION, NOT_STARTED, IN_PROGRESS, EXPIRED.